src/ImageSharp/ColorSpaces/Conversion/Implementation/Converters/HunterLabToCieXyzConverter.cs

```diff
@@ -26,9 +26,12 @@ public CieXyz Convert(in HunterLab input)
             float ka = ComputeKa(input.WhitePoint);
             float kb = ComputeKb(input.WhitePoint);
 
-            float y = ImageMaths.Pow2(l / 100F) * yn;
-            float x = (((a / ka) * MathF.Sqrt(y / yn)) + (y / yn)) * xn;
-            float z = (((b / kb) * MathF.Sqrt(y / yn)) - (y / yn)) * (-zn);
+            float pow = ImageMaths.Pow2(l / 100F);
+            float sqrtPow = MathF.Sqrt(pow);
+            float y = pow * yn;
+
+            float x = (((a / ka) * sqrtPow) + pow) * xn;
```

Comment:
Are you fixing a bug here or creating one? `sqrtPow != MathF.Sqrt(y / yn)` and `pow != y / yn`.

Reply:
`pow = ImageMaths.Pow2(l/100F)` // new intermediate variable
y was `ImageMaths.Pow2(l/100F) * yn`. Replace the first term by pow and you get
`y = pow*yn` // new assignment to y, line 31.

for x:
original: `x = (((a/ka) * Sqrt(y/yn)) + (y/yn)) * xn`
as pow is the old `y/yn` (that's why for the new y we multiply `pow * yn` now), we can replace this to
tmp1: `x = (((a/ka) * Sqrt(pow)) + pow) * xn`
as Sqrt(pow) is needed again, we replace that as well, so we end up with what I defined.

for z the same pattern as for x applies.

Or in Short according to your remark:
```
sqrtPow  
= MathF.Sqrt(pow) 
= MathF.Sqrt(ImagePath.Pow2(l/100)) 
= MathF.Sqrt(ImageMath.Pow2(l/100)*yn / yn)) 
= MathF.Sqrt(y/yn)
```

Reply:
That is a good catch. And thanks for clearing it up. Missed that you optimized the division away.